### PR TITLE
Update lavalink-with-ssl.md

### PR DESCRIPTION
**Pull request by MizaelS**
Add new host

### CODE REVIEW

**Comment by DarrenOfficial** [comment]
This server does not have an SSL certificate.

```bash
[~] : curl -v "https://lavalink.teramont.net:25565/"                                                                                                                                                                                04:32:50
* processing: https://lavalink.teramont.net:25565/
*   Trying 199.127.63.164:25565...
* Connected to lavalink.teramont.net (199.127.63.164) port 25565
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/pki/tls/certs/ca-bundle.crt
*  CApath: none
* OpenSSL/3.1.1: error:0A00010B:SSL routines::wrong version number
* Closing connection
curl: (35) OpenSSL/3.1.1: error:0A00010B:SSL routines::wrong version number
```

Please move it to lavalink-without-ssl

**Comment by MizaelS** [comment]
root@ubuntu:~# curl -v "https://lavalink.teramont.net/"
*   Trying 199.127.63.164:443...
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=lavalink.teramont.net
*  start date: Jan 19 02:54:45 2024 GMT
*  expire date: Apr 18 02:54:44 2024 GMT
*  subjectAltName: host "lavalink.teramont.net" matched cert's "lavalink.teramont.net"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x55c729090630)
> GET / HTTP/2
> Host: lavalink.teramont.net
> user-agent: curl/7.68.0
> accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 401 
< server: nginx/1.18.0 (Ubuntu)
< date: Fri, 19 Jan 2024 04:38:07 GMT
< content-length: 0
< lavalink-api-version: 3
< 
* Connection #0 to host lavalink.teramont.net left intact
root@ubuntu:~# 


The connection is then forwarded to port 25565 by nginx.

**Comment by MizaelS** [comment]
```
root@ubuntu:~# curl -v "https://lavalink.teramont.net:443/"
*   Trying 199.127.63.164:443...
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=lavalink.teramont.net
*  start date: Jan 19 02:54:45 2024 GMT
*  expire date: Apr 18 02:54:44 2024 GMT
*  subjectAltName: host "lavalink.teramont.net" matched cert's "lavalink.teramont.net"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x55c80a3c3630)
> GET / HTTP/2
> Host: lavalink.teramont.net
> user-agent: curl/7.68.0
> accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 401 
< server: nginx/1.18.0 (Ubuntu)
< date: Fri, 19 Jan 2024 04:50:07 GMT
< content-length: 0
< lavalink-api-version: 3
< 
* Connection #0 to host lavalink.teramont.net left intact
root@ubuntu:~# 
```
Like 443, but in this case it is forwarded to 25565.

**Comment by DarrenOfficial** [comment]
Change the port to 443 then.

File: docs/SSL/lavalink-with-ssl.md [hunk 1]
```diff
@@ -15,6 +15,17 @@ description: SSL Uses Secure WS connection, whilst No SSL uses standard WS. if y
     v4 introduce a breaking changes that affects all library that are using v3 API.
     You must update your bot library to support v4!
 
+### Hosted by @ [Teramont Host](https://www.teramont.net/)
+Version 3.7.10
+
+[Discord Server](https://www.teramont.net/discord)
+```bash
+Host: lavalink.teramont.net
+Port : 25565
```
Comment (DarrenOfficial) [inline docs/SSL/lavalink-with-ssl.md:24]: 443

**Review by DarrenOfficial — CHANGES_REQUESTED** [review]

**Comment by MizaelS** [comment]
It is not possible, this Lavalink is in docker and it is not allowed to start in the ports used by nginx (80 and 443). 
Likewise, players recognize it as secure using 25565

**Comment by DarrenOfficial** [comment]
> It is not possible, this Lavalink is in docker and it is not allowed to start in the ports used by nginx (80 and 443).
Likewise, players recognize it as secure using 25565

I mean on here, instead of the port 25565, change it to 443.

**Comment by MizaelS** [comment]
> > It is not possible, this Lavalink is in docker and it is not allowed to start in the ports used by nginx (80 and 443).
> > Likewise, players recognize it as secure using 25565
> 
> I mean on here, instead of the port 25565, change it to 443.

By changing it to 443 the connection will not reach the lavalink server.

**Comment by MizaelS** [comment]
```
server {
    listen 80;
    server_name lavalink.teramont.net;
    return 301 https://$server_name$request_uri;
}

server {
    listen 443 ssl http2;
    server_name lavalink.teramont.net;

    ssl_certificate /srv/server_certs/lavalink.teramont.net/cert.pem;
    ssl_certificate_key /srv/server_certs/lavalink.teramont.net/key.pem;
    ssl_session_cache shared:SSL:10m;
    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384";
    ssl_prefer_server_ciphers on;

    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_set_header Upgrade $http_upgrade;  
        proxy_set_header Connection "upgrade";   
        proxy_pass http://199.127.63.164:25565;
        proxy_http_version 1.1;                 
        proxy_read_timeout 90s;                
    }
}
```

Attached is the nginx configuration

**Comment by DarrenOfficial** [comment]
Looks like it's reaching the server just fine,

```bash
[~] : curl -v "https://lavalink.teramont.net/version" -H "Authorization: eHKuFcz67k4lBS64"                                                                                                                                          05:09:49
* processing: https://lavalink.teramont.net/version
*   Trying 199.127.63.164:443...
* Connected to lavalink.teramont.net (199.127.63.164) port 443
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/pki/tls/certs/ca-bundle.crt
*  CApath: none
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=lavalink.teramont.net
*  start date: Jan 19 02:54:45 2024 GMT
*  expire date: Apr 18 02:54:44 2024 GMT
*  subjectAltName: host "lavalink.teramont.net" matched cert's "lavalink.teramont.net"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* using HTTP/2
* h2 [:method: GET]
* h2 [:scheme: https]
* h2 [:authority: lavalink.teramont.net]
* h2 [:path: /version]
* h2 [user-agent: curl/8.2.1]
* h2 [accept: */*]
* h2 [authorization: eHKuFcz67k4lBS64]
* Using Stream ID: 1
> GET /version HTTP/2
> Host: lavalink.teramont.net
> User-Agent: curl/8.2.1
> Accept: */*
> Authorization: eHKuFcz67k4lBS64
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
< HTTP/2 200 
< server: nginx/1.18.0 (Ubuntu)
< date: Fri, 19 Jan 2024 05:09:54 GMT
< content-type: text/plain;charset=ISO-8859-1
< content-length: 6
< lavalink-api-version: 3
< 
* Connection #0 to host lavalink.teramont.net left intact
3.7.10
```

**Comment by DarrenOfficial** [comment]
> By changing it to 443 the connection will not reach the lavalink server.

I'm not talking about docker / application.yml, I'm talking the port specified here.

**Comment by DarrenOfficial** [comment]
Because here you mentioned the port 25565, which is not the port that client can use:

`Port : 25565`

```bash
[~] : curl -v "https://lavalink.teramont.net:25565/version" -H "Authorization: eHKuFcz67k4lBS64"                                                                                                                                    05:09:54
* processing: https://lavalink.teramont.net:25565/version
*   Trying 199.127.63.164:25565...
* Connected to lavalink.teramont.net (199.127.63.164) port 25565
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/pki/tls/certs/ca-bundle.crt
*  CApath: none
* OpenSSL/3.1.1: error:0A00010B:SSL routines::wrong version number
* Closing connection
curl: (35) OpenSSL/3.1.1: error:0A00010B:SSL routines::wrong version number
```

**Comment by MizaelS** [comment]
Ready, it has been modified to 443

**Review by DarrenOfficial — APPROVED** [review]

**Comment by DarrenOfficial** [comment]
Thanks for your contribution 👍🏼

**Pull request merged**